script/build/android-toolchain

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+
+# Adapted from build/android/make-toolchain
+
+set -e
+
+if [ $# -ne 2 -a $# -ne 3 ]; then
+    echo "" 1>&2
+    echo "Usage: $0 NDK_DIR ARCH [API]" 1>&2
+    echo "" 1>&2
+    echo "  NDK_DIR: path where NDK is installed" 1>&2
+    echo "  ARCH: armeabi, armeabi-v7a, arm64, mips, mips64, x86, x86_64" 1>&2
+    echo "  API: 1 ... 25" 1>&2
+    echo "" 1>&2
+    echo "If API is omitted, the minimum possible API for the " 1>&2
+    echo "required architecture is used." 1>&2
+    echo "" 1>&2
+    echo "Example usage (on MacOS with Android Studio installed):" 1>&2
+    echo "  $0 ~/Library/Android/sdk/ndk-bundle arm 19" 1>&2
+    echo "" 1>&2
+    exit 1
+fi
+
+ROOTDIR=$(cd $(dirname $0)/../../ && pwd -P)
+
+NDK_DIR=$1
+ARCH=$2
+API=$3
+BASEDIR=$ROOTDIR/MK_BUILD/android/toolchain
+
+# Directory where to install the toolchain. This must be done _before_
+# mapping the input architecture to a real toolchain architecture.
+INSTALL_DIR=$BASEDIR/${ARCH}
+
+# Map input architecture to toolchain architecture (armeabi and
+# armeabi-v7a both use the arm toolchain)
+if [ "$ARCH" = "armeabi" -o "$ARCH" = "armeabi-v7a" ]; then
+    ARCH="arm"
+fi
+
+if [ "$API" = "" ]; then
+    if [ "$ARCH" = "arm64" -o "$ARCH" = "mips64" -o "$ARCH" = "x86_64" ]; then
+        API=21
+    else
+        API=14
+    fi
```

Comment:
Is some comment on default API level useful here? Why is it 14-by-default and 21 for some platforms?

Reply:
Yes, I can add a comment. Basically: 14 is the minimum for 32 bit and 21 the minimum for 64 bit, and these limits are enforced by the NDK script for building the standalone toolchain. I do not remember whether, if passed a too low value, such script bails or just produces a toolchain that does not work. Anyway, I can probably find an issue or PR where I fixed this and point to that.

Reply:
Look, I've figured out how I've discovered these numbers:

```
$ git diff
diff --git a/script/build/android-toolchain b/script/build/android-toolchain
index 2a869652..65779959 100755
--- a/script/build/android-toolchain
+++ b/script/build/android-toolchain
@@ -40,7 +40,7 @@ fi
 
 if [ "$API" = "" ]; then
     if [ "$ARCH" = "arm64" -o "$ARCH" = "mips64" -o "$ARCH" = "x86_64" ]; then
-        API=21
+        API=14
     else
         API=14
     fi
$ ./script/build/android-toolchain ~/Library/Android/sdk/ndk-bundle arm64
Creating toolchain for API 14 and ARCH arm64 in /Users/sbasso/src/github.com/measurement-kit/measurement-kit/MK_BUILD/android/toolchain/arm64
14 is less than minimum platform for arm64 (21)
$ echo $?
1
```

I am going to reference this comment in the code :-).